test/functional/page_objects/common_page.js

```diff
@@ -28,129 +28,92 @@ export function CommonPageProvider({ getService, getPageObjects }) {
      * @param {string} appName As defined in the apps config
      * @param {string} subUrl The route after the hash (#)
      */
-    navigateToUrl(appName, subUrl) {
+    async navigateToUrl(appName, subUrl) {
       const appConfig = {
         ...config.get(['apps', appName]),
         // Overwrite the default hash with the URL we really want.
         hash: `${appName}/${subUrl}`,
       };
 
       const appUrl = getUrl.noAuth(config.get('servers.kibana'), appConfig);
-      return remote.get(appUrl);
+      await remote.get(appUrl);
+      await this.loginIfPrompted(appUrl);
+    }
+
+    async loginIfPrompted(appUrl) {
+      let currentUrl = await remote.getCurrentUrl();
+      log.debug(`currentUrl = ${currentUrl}\n    appUrl = ${appUrl}`);
+      await remote.setFindTimeout(defaultTryTimeout * 2).findByCssSelector('[data-test-subj="kibanaChrome"]');
+      const loginPage = currentUrl.includes('/login');
+      const wantedLoginPage = appUrl.includes('/login') || appUrl.includes('/logout');
+
+      // TODO: add the SAML login feature from integration-test repo here
```

Comment:
Is this going to happen in this PR or later one?

Reply:
I wasn't planning on doing it in this PR.  We already have some SAML Tests that are not UI tests so I'm not sure we'll ever add any here.  And we do have an integration-test that already does it.

Reply:
> so I'm not sure we'll ever add any here.

So does that mean this TODO is unnecessary? Or do you mean we'll never add any in this PR?

Yeah we have some SAML API tests in the functional_tests script for x-pack. Is that sufficient for cloud? I wonder if there will ever be anything UI related around SAML, but I doubt it.